src/v/raft/consensus.cc

```diff
@@ -517,37 +517,29 @@ ss::future<result<model::offset>> consensus::linearizable_barrier() {
     co_return ret_t(snapshot.linearizable_offset);
 }
 
-ss::future<result<replicate_result>>
+replicate_stages
 consensus::replicate(model::record_batch_reader&& rdr, replicate_options opts) {
-    return ss::with_gate(_bg, [this, rdr = std::move(rdr), opts]() mutable {
-        return do_replicate({}, std::move(rdr), opts);
-    });
+    return do_replicate({}, std::move(rdr), opts);
 }
 
-ss::future<result<replicate_result>> consensus::replicate(
+replicate_stages consensus::replicate(
   model::term_id expected_term,
   model::record_batch_reader&& rdr,
   replicate_options opts) {
-    return ss::with_gate(
-      _bg, [this, rdr = std::move(rdr), opts, expected_term]() mutable {
-          return do_replicate(expected_term, std::move(rdr), opts);
-      });
+    return do_replicate(expected_term, std::move(rdr), opts);
 }
```

Comment:
Can you add a comment in the commit message about why this gate is no longer taken, it doesn't seem to go with the rest of the commit, which is pure refactor.